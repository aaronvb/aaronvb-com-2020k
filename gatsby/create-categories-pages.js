'use strict';

const _ = require('lodash');
const path = require('path');

module.exports = async (graphql, actions) => {
  const { createPage } = actions;

  const result = await graphql(`
    {
      allMarkdownRemark(
        filter: { frontmatter: { template: { eq: "post" }, draft: { ne: true } } }
      ) {
        group(field: frontmatter___category) {
          fieldValue
          totalCount
        }
      }
    }
  `);

  _.each(result.data.allMarkdownRemark.group, (category) => {
    const categorySlug = `/category/${_.kebabCase(category.fieldValue)}`;

    createPage({
      path: categorySlug,
      component: path.resolve('./src/templates/category-template.js'),
      context: {
        category: category.fieldValue,
      }
    });
  });
};
