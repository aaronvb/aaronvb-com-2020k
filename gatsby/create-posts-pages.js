'use strict';

const path = require('path');

module.exports = async (graphql, actions) => {
  const { createPage } = actions;

  createPage({
    path: '/',
    component: path.resolve('./src/templates/index-template.js'),
  });
};
