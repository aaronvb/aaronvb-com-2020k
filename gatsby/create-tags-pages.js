'use strict';

const _ = require('lodash');
const path = require('path');

module.exports = async (graphql, actions) => {
  const { createPage } = actions;

  const result = await graphql(`
    {
      allMarkdownRemark(
        filter: { frontmatter: { template: { eq: "post" }, draft: { ne: true } } }
      ) {
        group(field: frontmatter___tags) {
          fieldValue
          totalCount
        }
      }
    }
  `);

  _.each(result.data.allMarkdownRemark.group, (tag) => {
    const tagSlug = `/tag/${_.kebabCase(tag.fieldValue)}`;

    createPage({
      path: tagSlug,
      component: path.resolve('./src/templates/tag-template.js'),
      context: {
        tag: tag.fieldValue,
      }
    });
  });
};
