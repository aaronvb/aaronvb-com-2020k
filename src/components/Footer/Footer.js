// @flow strict
import React from 'react';
import Copyright from './Copyright';
import Daily from './Daily';
import styles from './Footer.module.scss';
import { useSiteMetadata } from '../../hooks';

const Footer = () => {
  const { copyright } = useSiteMetadata();

  return (
    <div className={styles['footer']}>
      <div className={styles['footer__inner']}>
        <div className={styles['footer__copyright']}>
          <Copyright copyright={copyright} />
        </div>
        <div className={styles['footer__daily']}>
          <Daily />
        </div>
      </div>
    </div>
  );
};

export default Footer;
