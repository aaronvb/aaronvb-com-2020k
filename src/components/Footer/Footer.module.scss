@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';

.footer {
    max-width: $layout-post-width;
    margin-left: auto;
    margin-right: auto;
    &__inner {
        padding: 0 20px 0;
        @include margin-top(2);
        @include margin-bottom(1);
    }
}

@include breakpoint-sm {
    .footer {
        &__inner {
            padding: 0;
            @include margin-top(2);
            @include margin-bottom(1);
        }
    }
}

@include breakpoint-md {
    .footer {
        &__inner {
            padding: 0;
            @include margin-top(2);
            @include margin-bottom(1);
        }
        &__copyright {
            lost-column: 1/2;
        }
        &__daily {
            text-align: right;
        }
    }
}
