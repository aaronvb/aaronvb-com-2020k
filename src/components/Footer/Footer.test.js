// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import MockDate from 'mockdate';
import Footer from './Footer';
import siteMetadata from '../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../types';

describe('Tobar', () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(siteMetadata)
      ),
      useStaticQuery.mockReturnValue(siteMetadata)
    );
  });

  const props = {
    isIndex: true
  };

  it('renders correctly', () => {
    MockDate.set(new Date('2020-04-30T11:01:58.135Z'));
    const tree = renderer.create(<Footer {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
    MockDate.reset();
  });
});
