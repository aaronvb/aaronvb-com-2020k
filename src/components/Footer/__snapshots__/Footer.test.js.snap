// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tobar renders correctly 1`] = `
<div
  className="footer"
>
  <div
    className="footer__inner"
  >
    <div
      className="footer__copyright"
    >
      <div
        className="copyright"
      >
        Test copyright
      </div>
    </div>
    <div
      className="footer__daily"
    >
      <div
        className="daily"
      >
        Happy Thursday. â˜•
      </div>
    </div>
  </div>
</div>
`;
